react/food-app/src/App.js

```diff
@@ -1,15 +1,53 @@
-import React, { useEffect, useState } from 'react';
-import Foodcard from './components/foodCard'
-import './App.css';
+import React, { useEffect, useState } from "react";
+import Foodcard from "./components/foodCard";
+import "./App.css";
 
 function App() {
-return <div>
-  <Foodcard/>
-</div> 
-}
-
+  const [food, setFood] = useState([]);
+  const [searchTerm, setSearchTerm] = useState("");
 
+  const onChange = (e) => {
+    setSearchTerm(e.target.value);
+  };
+  const onSubmit = (e) => {
+    e.preventDefault();
+  };
+  useEffect(() => {
+    fetch(
+      "https://pair-programming-test.s3.eu-west-2.amazonaws.com/cuisine.json"
+    )
+      .then((res) => res.json())
+      .then((data) => setFood(data));
+  }, []);
 
+  return (
+    <div>
+      <form onSubmit={onSubmit} className="form">
+        <label>Search For Your Favourite Food</label>
+        <input
+          className="input-search"
+          onChange={onChange}
+          type="text"
+          placeholder="UK, Fish, Pizza ..."
+        />
+      </form>
 
+      <ul className="food-ul">
+        {food.map((item) => {
+          if (searchTerm) {
+            if (
+              item.name.includes(searchTerm) ||
+              item.origin.includes(searchTerm)
+            ) {
+              return <Foodcard food={item} key={item.id} />;
+            }
```

Comment:
This is quite a cheeky condition because it implicitly means the map's predicate doesn't return anything if there's a search term but it doesn't match a particular item. More explicit code is generally more readable code. 

My challenge to you is to find an array method that you can call on food before `map` that will remove the items that don't match your search term before you render them

Reply:
Hey Mark, I just made some changes to the following that you suggested. i just commited it. 
Thanks